spec/spec_helper.rb

```diff
@@ -19,28 +19,38 @@
 
   DynamoDBMutex::Lock.logger = Logger.new(log_stream)
 
-  dynamo_thread = nil
+  dynamo_pid = nil
+
+  config.before(:all) do
+    dynamo_pid = Process.fork do
+      Dir.chdir('resources')
+      $stdout.reopen('/dev/null', 'w')
+
+      exec('java -Djava.library.path=./DynamoDBLocal_lib -jar DynamoDBLocal.jar -inMemory -port 4567')
+    end
 
-  config.before(:suite) do
-    FakeDynamo::Logger.setup(:debug)
-    FakeDynamo::Storage.instance.init_db('test.fdb')
-    FakeDynamo::Storage.instance.load_aof
+    # Loop until we successfully connect to the DynamoDB
+    # maximal loop waiting time in seconds
+    loop_max_wait = 10
+    loop_count = 0
+    sleep_interval = 0.1
 
-    dynamo_thread = Thread.new do
-      $stdout = log_stream # Throw FakeDynamo's stdout logging somewhere else.
-      FakeDynamo::Server.run!(port: 4567, bind: 'localhost') do |server|
-        if server.respond_to?('config') && server.config.respond_to?('[]=')
-          server.config[:AccessLog] = []
-        end
+    while loop_count < loop_max_wait / sleep_interval
+      begin
+        client = Aws::DynamoDB::Client.new
+        client.list_tables
+
+        break
+      rescue
+        sleep sleep_interval
       end
     end
-    sleep(1)
-  end
 
-  config.after(:suite) do
-    FakeDynamo::Storage.instance.shutdown
-    dynamo_thread.exit if dynamo_thread
-    FileUtils.rm('test.fdb', force: true)
+    raise 'DynamoDB did not start in time' if loop_count >= loop_max_wait / sleep_interval
```

Comment:
What about something as simple as this?

```ruby
[0.1, 0.5, 1, 5, 10].map do |sleep_interval|
  Aws::DynamoDB::Client.new.list_tables rescue (sleep sleep_interval; false)
end.any? || fail('DynamoDB did not start in time')
```

Notice that it'll implicitly break 🙂

Reply:
`#golfing` :1st_place_medal: 

Will it implicitly break? Wont `[..].map <block>` evaluate the block for each value? What am I missing?

Reply:
Darn, I'm missing a `.lazy`:

```ruby
[0.1, 0.5, 1, 5, 10].lazy.map do |sleep_interval|
```

Reply:
Pretty sweet! :+1: